driver-core/src/test/java/com/datastax/driver/core/AggregateMetadataTest.java

```diff
@@ -43,6 +44,13 @@ public void setup() {
     }
 
     @Test(groups = "unit")
+    public void should_use_all_fields_in_equals_and_hashCode() {
+        EqualsVerifier.forClass(AggregateMetadata.class)
+            .allFieldsShouldBeUsedExcept("simpleName", "finalFuncSimpleName", "stateFuncSimpleName", "stateTypeCodec")
```

Comment:
simple names are derived from other fields so don't need to be included, codec doesn't seem relevant either.